### ISSUE TITLE
Early exit for CheckCorrelatedFeatures (SKD008) for high-dimensional datasets

### ISSUE THREAD

**Issue by yashvisharma1204** [probabl-ai/skore#2949]
### Describe the bug

The bottleneck occurs in CheckCorrelatedFeatures (SKD008). The check uses spearmanr(X), which calculates an $N \times N$ dense correlation matrix. For a dataset with 50,000 features, this requires allocating roughly 20 GB of contiguous RAM in a single block

**Proposed Improvement:**
Instead of simply skipping the check for wide datasets, we can randomly sample a subset of features (e.g., `MAX_FEATURES = 1000` with a fixed `seed=42`). This caps the memory footprint at `<10MB` while still statistically retaining the ability to detect severe multicollinearity, adding performance resiliency without losing the value of the check.

*(Note: I have a PR ready with this exact sampling improvement!)*

### Steps/Code to Reproduce

import numpy as np
from sklearn.dummy import DummyClassifier
from skore import evaluate

# 1. Define dimensions (50k features triggers a ~20GB allocation attempt)
n_samples = 500
n_features = 50000 

print(f"Generating synthetic dataset: {n_samples} rows, {n_features} features...")
X = np.random.rand(n_samples, n_features)
y = np.random.randint(0, 2, n_samples)

report = evaluate(DummyClassifier(), X, y, splitter=0.2)

# 2. This triggers SKD008 and causes the OOM crash on standard machines
print("Running checks...")
summary = report.checks.summarize()

### Expected Behavior

The check should recognize that the dataset is extremely wide, randomly sample 1,000 features (using a fixed seed for determinism), and perform the correlation check on that subset. This prevents the crash, allowing the rest of the pipeline to finish successfully while still warning the user if multicollinearity is detected in the sample.

### Actual Behavior

The `spearmanr` function attempts to allocate a massive contiguous block of memory for the correlation matrix, resulting in a `MemoryError` and completely crashing the Python process/notebook.

### Environment

```shell
System:
    python: 3.12.13 (main, Mar  4 2026, 09:23:07) [GCC 11.4.0]
executable: /usr/bin/python3
   machine: Linux-6.6.122+-x86_64-with-glibc2.35

Python dependencies:
        skore: 0.18.1
          pip: 24.1.2
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.0
        numpy: 2.0.2
       pandas: 2.2.2
       plotly: 5.24.1
         rich: 15.0.0
 scikit-learn: 1.6.1
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.9.0
skore-hub-project: None
xgboost-cpu; (platform_system: None
xgboost; (platform_system: None
```

**Comment by glemaitre** [comment]
Could you describe you're use case where you have 50,000 features?

**Comment by yashvisharma1204** [comment]
> Could you describe you're use case where you have 50,000 features?

Thanks for reviewing !We are using a transformer model to extract text embeddings and then train a standard Scikit-Learn model on those embeddings. By the time we pass the data into skore, it has over 50,000 dense features which leads to massive memory allocation

**Comment by glemaitre** [comment]
OK, I see. So it makes sense to have guardrails then.

I'm even thinking that you should be ignoring via the config the check because it should not be relevant in your case.

A potential case where we should think about it is in the case that you have still many feature like DNA sequences and see how we could manage in this case. Here, it would still be relevant to check the info.
